File: backend/src/main/kotlin/com/millo/hayoungplace/place/domain/Comment.kt
package com.millo.hayoungplace.place.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(collection = "comments")
data class Comment(
    @Id
    val id: String? = null,
    val placeId: String,
    val parentId: String? = null, // 대댓글의 경우 부모 댓글 ID
    val nickname: String,
    val password: String, // 암호화된 패스워드
    val content: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
    val isDeleted: Boolean = false // 소프트 삭제
)

File: frontend/src/types/comment.ts
// 댓글 작성 요청
export interface CommentCreateRequest {
  nickname: string;
  password: string;
  content: string;
  parentId?: string;
}

// 댓글 수정 요청
export interface CommentUpdateRequest {
  password: string;
  content: string;
}

// 댓글 삭제 요청
export interface CommentDeleteRequest {
  password: string;
}

// 댓글 응답
export interface CommentResponse {
  id: string;
  placeId: string;
  parentId?: string;
  nickname: string;
  content: string;
  createdAt: string;
  updatedAt: string;
  isDeleted: boolean;
  replies: CommentResponse[];
}

// 댓글 페이지 응답
export interface CommentPageResponse {
  content: CommentResponse[];
  totalPages: number;
  totalElements: number;
  currentPage: number;
  size: number;
} 